docs-src/guide-rpc-connect.scrbl

```diff
@@ -0,0 +1,519 @@
+        if ( v instanceof Function ) {
+          meths[k] = true;
+        } else {
+          vals[k] = v;
+        }
+      }
+      return new Promise((resolve, reject) => (async () => {
+        let p = rpc(m, arg, vals, meths);
+        while (true) {
+          try {
+            const r = await p;
+            switch ( r.t ) {
+              case 'Done': {
+                return resolve(r.ans);
+              }
+              case 'Kont': {
+                const { kid, m, args } = r;
+                const ans = await cbacks[m](...args);
+                p = rpc(`/kont`, kid, ans);
+                break;
+              }
+              default:
+                throw new Error(`Illegal callback return: ${JSON.stringify(r)}`);
+            }
+          } catch (e) {
+            return reject(e);
+          }
+        }
+      })());
+    };
+
+    return { rpc, rpcReady, rpcCallbacks };
+  };
+}
+
+Let's break this down.
+
+
+@;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+@subsection[#:tag "rpc-connect-js-lib-walkthrough-imports"]{Imports}
+
+First our imports:
+
+@js{
+  import waitPort from 'wait-port';
+  import bent     from 'bent';
+}
+
+@jsin{waitPort} is how our client will pause and wait for the server to become
+available. If there isn't already a canned solution to this problem in your
+language it's simple to
+@seclink["rpc-connect-await-online"]{implement one yourself}.
+
+@jsin{bent} handles our HTTPS requests, as you'll see in a moment.
+
+
+@;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+@subsection[#:tag "rpc-connect-js-lib-walkthrough-options"]{Options}
+
+Next some admin:
+
+@js{
+  export const mkRPC = opts => {
+    const defaults = {
+      host:   'REACH_RPC_SERVER',
+      port:   'REACH_RPC_PORT',
+      apiKey: 'REACH_RPC_KEY',
+    };
+
+    const r = (acc, k) =>
+      Object.assign(acc, { [k]: process.env[defaults[k]] });
+
+    const o = Object.assign(
+      Object.keys(defaults).reduce(r, {}),
+      opts);
+
+    Object.keys(o).forEach(k => {
+      if (!o[k]) {
+        throw new Error(`Neither \`opts.${k}\` nor ${defaults[k]} environment`
+                      + ` variable are configured!`);
+      }
+    });
+
+  ...
+  };
+}
+
+The client defaults to using the @envvar{REACH_RPC_<OPTION NAME>} environment
+variables mentioned earlier but allows consuming code to selectively override
+these values. If neither an @jsin{opts['<option name>']} nor corresponding
+environment variable has been populated then an exception is thrown and the
+library-user will be unable to continue.
+
+
+@;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+@subsection[#:tag "rpc-connect-js-lib-walkthrough-post"]{HTTP POST}
+
+Next we'll create a shorthand for actually sending @tech{RPC} requests.
+
+@js{
+  const call = bent(`https://${o.host}:${o.port}`, `POST`, `json`, 200, {
+    'X-API-Key': o.apiKey,
```

Comment:
This section makes me think that even this section needs to be split again into one section that documents the server protocol and another that talks about the implementation of a client... then this part right here can link to the corresponding section in the protocol docs about how you need an api key